src/app/redux/reducer.ts

```diff
@@ -12,7 +13,7 @@ const initialState: StockState = {
 };
 
 export const stockReducer: Reducer<
-  StockState,
+  Readonly<StockState>,
```

Comment:
👍 for making all the reducer functions use `Readonly` state. 

It would be better to include this in the type definition: 

```ts
export type StockState = {
  readonly selectedStock: StockProps | undefined;
};
```

Reply:
Good point!  will do it in another PR